src/overlays/actors/ovl_En_Kgy/z_en_kgy.c

```diff
@@ -41,105 +63,1132 @@ const ActorInit En_Kgy_InitVars = {
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/func_80B41C30.s")
+s32 func_80B42FA0(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnKgy* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/func_80B41C54.s")
+    if (!(this->unk_29C & 1)) {
+        if (limbIndex == 17) {
+            *dList = NULL;
+        }
+    } else if (limbIndex == 16) {
+        *dList = NULL;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/func_80B41CBC.s")
+    if (limbIndex == 11) {
+        rot->x += this->unk_2CC.y;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/func_80B41D64.s")
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/func_80B41E18.s")
+void func_80B43008(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    static Vec3f D_80B432D8 = { 1000.0f, 2000.0f, 0.0f };
+    static Vec3f D_80B432E4 = { 3000.0f, 4000.0f, 300.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/func_80B42508.s")
+    EnKgy* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/func_80B425A0.s")
+    if (limbIndex == 11) {
+        SysMatrix_MultiplyVector3fByState(&D_80B432D8, &this->unk_2A8);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/func_80B42660.s")
+    if (limbIndex == 16) {
+        SysMatrix_MultiplyVector3fByState(&D_80B432E4, &this->unk_2C0);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/func_80B42714.s")
+void func_80B43074(EnKgy* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Gfx* gfx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/func_80B427C8.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/func_80B4296C.s")
+    func_8012C28C(globalCtx->state.gfxCtx);
+    func_800B8050(&this->actor, globalCtx, MTXMODE_NEW);
+    SysMatrix_StatePush();
+    SysMatrix_InsertTranslation(-800.0f, 3100.0f, 8400.0f, MTXMODE_APPLY);
+    SysMatrix_InsertXRotation_s(0x4000, MTXMODE_APPLY);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/func_80B42A8C.s")
+    if (func_80B40D8C(globalCtx)) {
+        AnimatedMat_Draw(globalCtx, Lib_SegmentedToVirtual(D_0600F6A0));
+    } else {
+        AnimatedMat_Draw(globalCtx, Lib_SegmentedToVirtual(D_0600EE58));
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/func_80B42D28.s")
+    gfx = POLY_OPA_DISP;
+    gSPMatrix(gfx, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kgy/EnKgy_Update.s")
+    if (func_80B40D8C(globalCtx)) {
+        gSPDisplayList(gfx + 1, D_04001D00);
+        gSPDisplayList(gfx + 2, D_0600F180);
+    } else {
+        gSPDisplayList(gfx + 1, D_040021A8);
+        gSPDisplayList(gfx + 2, D_0600E8F0);
+    }
+    POLY_OPA_DISP = gfx + 3;
```

Comment:
```suggestion
    if (func_80B40D8C(globalCtx)) {
        gSPDisplayList(&gfx[1], D_04001D00);
        gSPDisplayList(&gfx[2], D_0600F180);
    } else {
        gSPDisplayList(&gfx[1], D_040021A8);
        gSPDisplayList(&gfx[2], D_0600E8F0);
    }
    POLY_OPA_DISP = &gfx[3];
```
Maybe better if it matches. We should decide what is appropriate for this sort of thing in general.